nsxt/resource_nsxt_policy_project.go

```diff
@@ -143,7 +169,41 @@ func resourceNsxtPolicyProjectPatch(connector client.Connector, d *schema.Resour
 	log.Printf("[INFO] Patching Project with ID %s", id)
 
 	client := infra.NewProjectsClient(connector)
-	return client.Patch(defaultOrgID, id, obj)
+	err := client.Patch(defaultOrgID, id, obj)
+	if err != nil {
+		return err
+	}
+	return patchVpcSecurityProfile(d, connector, id)
+}
+
+func patchVpcSecurityProfile(d *schema.ResourceData, connector client.Connector, projectID string) error {
+	enabled := false
+	defaultSecurityProfile := d.Get("default_security_profile")
+	if defaultSecurityProfile != nil {
+		northSouthFirewall := defaultSecurityProfile.([]interface{})[0]
+		if northSouthFirewall != nil {
+			// This is safe as TF enforces
+			nsfw := northSouthFirewall.([]interface{})[0].(map[string]interface{})
+			enabled = nsfw["enabled"].(bool)
+		}
+	}
+	client := projects.NewVpcSecurityProfilesClient(connector)
+	_, err := client.Get(defaultOrgID, projectID, "default")
+
+	if isNotFoundError(err) && !enabled {
+		// No action required
```

Comment:
Are we dealing with the case in which default profile was not created yet? In this case, should we busy wait here?